FirebaseStorage/Tests/SwiftIntegration/StorageIntegration.swift

```diff
@@ -0,0 +1,678 @@
+        XCTFail("Failed to get snapshot.progress")
+        return
+      }
+      XCTAssertGreaterThanOrEqual(progress.completedUnitCount, uploadedBytes)
+      uploadedBytes = progress.completedUnitCount
+    })
+    waitForExpectations()
+  }
+
+  func testPutFileWithSpecialCharacters() throws {
+    let expectation = self.expectation(description: "testPutFileWithSpecialCharacters")
+
+    let fileName = "hello&+@_ .txt"
+    let ref = storage?.reference(withPath: "ios/public/" + fileName)
+    let data = try XCTUnwrap("Hello Swift World".data(using: .utf8), "Data construction failed")
+    let tmpDirURL = URL(fileURLWithPath: NSTemporaryDirectory())
+    let fileURL = tmpDirURL.appendingPathComponent("hello.txt")
+    do {
+      try data.write(to: fileURL, options: Data.WritingOptions.atomicWrite)
+    } catch {
+      XCTFail("Write to \(fileURL) failed")
+    }
+    ref?.putFile(from: fileURL, metadata: nil, completion: { metadata, error in
+      XCTAssertNotNil(metadata, "Metadata should not be nil")
+      XCTAssertNil(error, "Error should be nil")
+      XCTAssertEqual(fileName, metadata?.name)
+      ref?.getMetadata(completion: { metadata, error in
+        XCTAssertNotNil(metadata, "Metadata should not be nil")
+        XCTAssertNil(error, "Error should be nil")
+        XCTAssertEqual(fileName, metadata?.name)
+        expectation.fulfill()
+      })
+    })
+    waitForExpectations()
+  }
+
+  func testUnauthenticatedSimplePutDataNoMetadata() throws {
+    let expectation = self.expectation(description: "testUnauthenticatedSimplePutDataNoMetadata")
+
+    let ref = storage?.reference(withPath: "ios/public/testUnauthenticatedSimplePutDataNoMetadata")
+    let data = try XCTUnwrap("Hello Swift World".data(using: .utf8), "Data construction failed")
+    ref?.putData(data, metadata: nil, completion: { metadata, error in
+      XCTAssertNotNil(metadata, "Metadata should not be nil")
+      XCTAssertNil(error, "Error should be nil")
+      expectation.fulfill()
+    })
+    waitForExpectations()
+  }
+
+  func testUnauthenticatedSimplePutFileNoMetadata() throws {
+    let expectation = self.expectation(description: "testUnauthenticatedSimplePutFileNoMetadata")
+
+    let fileName = "hello&+@_ .txt"
+    let ref = storage?.reference(withPath: "ios/public/" + fileName)
+    let data = try XCTUnwrap("Hello Swift World".data(using: .utf8), "Data construction failed")
+    let tmpDirURL = URL(fileURLWithPath: NSTemporaryDirectory())
+    let fileURL = tmpDirURL.appendingPathComponent("hello.txt")
+    do {
+      try data.write(to: fileURL, options: Data.WritingOptions.atomicWrite)
+    } catch {
+      XCTFail("Write to \(fileURL) failed")
+    }
+    ref?.putFile(from: fileURL, metadata: nil, completion: { metadata, error in
+      XCTAssertNotNil(metadata, "Metadata should not be nil")
+      XCTAssertNil(error, "Error should be nil")
+      XCTAssertEqual(fileName, metadata?.name)
+      ref?.getMetadata(completion: { metadata, error in
```

Comment:
This seems to diverge from the original test,

Reply:
Fixed. Good catch!